src/components/jellyfinActions.ts

```diff
@@ -422,25 +422,35 @@ export function getDownloadSpeed(byteSize: number): Promise<any> {
     return JellyfinApi.authAjax(url, {
         type: 'GET',
         timeout: 5000
-    }).then(function () {
-        const responseTimeSeconds = (new Date().getTime() - now) / 1000;
-        const bytesPerSecond = byteSize / responseTimeSeconds;
-        const bitrate = Math.round(bytesPerSecond * 8);
-
-        return bitrate;
-    });
+    })
+        .then(function (response) {
+            // wait for the entire body to appear
+            return response.blob();
+        })
+        .then(function () {
+            const responseTimeSeconds = (new Date().getTime() - now) / 1000;
+            const bytesPerSecond = byteSize / responseTimeSeconds;
+            const bitrate = Math.round(bytesPerSecond * 8);
+
+            return bitrate;
+        });
```

Comment:
Any reason why this is not an arrow function?

Reply:
I'll make it one :)